frontend/src/components/maps/leaflet/InventoryLayer.tsx

```diff
@@ -80,14 +147,21 @@ export const InventoryLayer: React.FC<InventoryLayerProps> = ({
     throw new Error('<InventoryLayer /> must be used under a <Map> leaflet component');
   }
 
-  const bbox = getBbox(bounds);
+  const bbox = useMemo(() => getBbox(bounds), [bounds]);
+  useEffect(() => {
+    if (filterChanged) {
+      map.fitBounds(defaultBounds, { maxZoom: 5 });
+    }
+  }, [map, filterChanged]);
 
   minZoom = minZoom ?? 0;
   maxZoom = maxZoom ?? 18;
 
-  const params = useMemo<IGeoSearchParams>(
-    () => ({
-      bbox: getApiBbox(bounds ?? map.getBounds()).toString(),
+  const params = useMemo((): any => {
+    const tiles = getTiles(defaultBounds, 5);
```

Comment:
So whenever the filter changes we get all of tiles based on the default bounds again? wouldn't the result of that function call be the same every time and therefore be memoizable?